src/integration-tests/java/com/webdev/cosmo/cosmobackend/features/Oauth2IT.java

```diff
@@ -0,0 +1,66 @@
+package com.webdev.cosmo.cosmobackend.features;
+
+import com.webdev.cosmo.cosmobackend.BaseTestConfiguration;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.*;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.server.resource.authentication.BearerTokenAuthenticationToken;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider;
+
+import static org.mockito.Mockito.when;
+
+public class Oauth2IT extends BaseTestConfiguration {
```

Comment:
Rename into AuthorizationIT, we do not test directly Oauth2 flow as it is, but only small part of it 😄